lib/de.ml

```diff
@@ -2605,6 +2887,1050 @@ module Def = struct
+          then c.num_offset_syms <- num_offset_syms
+          else g (num_offset_syms - 1) in
+        g num_offset_syms
+        ; if c.num_litlen_syms <> num_litlen_syms then (
+            let max1 =
+              min (c.num_litlen_syms + c.num_offset_syms) num_litlen_syms
+              - c.num_litlen_syms in
+            let max2 = c.num_offset_syms - max1 in
+            for i = 0 to max1 - 1 do
+              c.codes.lens.litlen.(c.num_litlen_syms + i) <-
+                c.codes.lens.offset.(i)
+            done
+            ; for i = 0 to max2 - 1 do
+                c.codes.lens.offset.(i) <- c.codes.lens.offset.(max1 + i)
+              done)
+        ; c.num_precode_items <-
+            compute_precode_items
+              (Array.append c.codes.lens.litlen c.codes.lens.offset)
+              (c.num_litlen_syms + c.num_offset_syms)
+              c.precode_freqs c.precode_items
+        ; make_huffman_code num_precode_syms max_pre_codeword_len
+            c.precode_freqs c.precode_lens c.precode_codewords
+        ; let rec h num_explicit_lens =
+            if
+              num_precode_syms < 5
+              || c.precode_lens.(precode_lens_permutation.(num_explicit_lens - 1))
+                 <> 0
+            then c.num_explicit_lens <- num_explicit_lens
+            else h (num_explicit_lens - 1) in
+          h num_precode_syms
+
+          ; if c.num_litlen_syms <> num_litlen_syms then (
+              let max1 =
+                min (num_litlen_syms - c.num_litlen_syms) c.num_offset_syms
+              in
+              let max2 = max (c.num_offset_syms - max1) 0 in
+              for i = 0 to max2 - 1 do
+                c.codes.lens.offset.(max1 + max2 - 1 - i) <-
+                  c.codes.lens.offset.(max2 - 1 - i)
+              done
+              ; for i = 0 to max1 - 1 do
+                  c.codes.lens.offset.(i) <-
+                    c.codes.lens.litlen.(c.num_litlen_syms + i)
+                done)
+
+    let write_huffman_header c os =
+      add_bits os (c.num_litlen_syms - 257) 5
+      ; add_bits os (c.num_offset_syms - 1) 5
+      ; add_bits os (c.num_explicit_lens - 4) 4
+      ; flush_bits os
+      ; for i = 0 to c.num_explicit_lens - 1 do
+          add_bits os c.precode_lens.(precode_lens_permutation.(i)) 3
+          ; flush_bits os
+        done
+      ; for i = 0 to c.num_precode_items - 1 do
+          let precode_item = c.precode_items.(i) in
+          let precode_sym = precode_item land 0x1F in
+          add_bits os
+            c.precode_codewords.(precode_sym)
+            c.precode_lens.(precode_sym)
+          ; if precode_sym >= 16 then
+              if precode_sym == 16 then add_bits os (precode_item lsr 5) 2
+              else if precode_sym == 17 then add_bits os (precode_item lsr 5) 3
+              else add_bits os (precode_item lsr 5) 7
+          ; flush_bits os
+        done
+
+    let max_extra_length_bits = 5
+    let max_extra_offset_bits = 14
```

Comment:
To homogenize with the rest of the code, constant should be prepended by a `_` :+1:.

Reply:
This is done, however, I'll have to come back to them, as we have several doublons with the streamable api